cloudslang-content-verifier/src/main/java/io/cloudslang/lang/tools/build/validation/StaticValidatorImpl.java

```diff
@@ -0,0 +1,91 @@
+import io.cloudslang.lang.entities.bindings.InOutParam;
+import org.apache.commons.collections4.ListUtils;
+import org.apache.commons.lang.Validate;
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Created by bancl on 8/30/2016.
+ */
+@Component
+public class StaticValidatorImpl implements StaticValidator {
+
+    @Override
+    public void validateSlangFile(File slangFile, Executable executable, Metadata metadata, Boolean shouldValidateDescription){
+        validateNamespace(slangFile, executable);
+
+        validateExecutableName(slangFile, executable);
+
+        if (shouldValidateDescription) {
+            validateExecutableAgainstMetadata(executable, metadata);
+        }
+    }
+
+    private void validateExecutableAgainstMetadata(Executable executable, Metadata metadata) {
+        validateInOutParams(metadata.getInputs(), executable.getInputs(), "Error for executable " +
+                executable.getId() + ": Input");
+        validateInOutParams(metadata.getOutputs(), executable.getOutputs(), "Error for executable " +
+                executable.getId() + ": Output");
+        validateInOutParams(metadata.getResults(), executable.getResults(), "Error for executable " +
+                executable.getId() + ": Result");
+    }
+
+    private void validateInOutParams(Map<String, String> metadataInOutParams, List<? extends InOutParam> inOutParams, String errorMessagePrefix) {
+        for (InOutParam inOutParam : ListUtils.emptyIfNull(inOutParams)) {
+            if (metadataInOutParams == null) {
+                throw new RuntimeException(errorMessagePrefix + "s are missing description entirely.");
+            } else if (metadataInOutParams.get(inOutParam.getName()) == null) {
+                throw new RuntimeException(errorMessagePrefix + " '" + inOutParam.getName() + "' is missing description.");
+            }
+        }
+    }
+
+    private void validateNamespace(File slangFile, Executable executable) {
+        // Validate that the namespace is not empty
+        String namespace = executable.getNamespace();
+        Validate.notEmpty(namespace, "Error validating Slang file: \'" + slangFile.getAbsoluteFile() +
+                "\'. Namespace of slang source: \'" + executable.getName() + "\' cannot be empty.");
+
+        // Validate that the namespace matches the path of the file
+        String executableNamespacePath = namespace.replace('.', File.separatorChar);
+        String namespaceErrorMessage = "Error validating Slang file: \'" + slangFile.getAbsoluteFile() +
+                "\'. Namespace of slang source: " + executable.getName() + " is wrong.\nIt is currently \'" +
+                namespace + "\', but it should match the file path: \'" + slangFile.getPath() + "\'";
+        int indexOfLastFileSeparator = slangFile.getAbsolutePath().lastIndexOf(File.separatorChar);
```

Comment:
you can use something like `new File(executableNamespacePath).getParent().getAbsolutePath()`